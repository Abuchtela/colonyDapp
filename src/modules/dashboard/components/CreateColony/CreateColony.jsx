/* @flow */

export { default } from '../../../pages/components/WizardTemplate';
