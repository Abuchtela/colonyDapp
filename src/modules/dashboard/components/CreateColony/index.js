/* @flow */

export { default } from './CreateColony';
