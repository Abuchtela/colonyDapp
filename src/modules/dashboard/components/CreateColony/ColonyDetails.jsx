/* @flow */

import React from 'react';
import { defineMessages, FormattedMessage } from 'react-intl';

import type { FormProps } from '~types/forms';

import layout from '~styles/layout.css';
import styles from './ColonyDetails.css';

import { FieldSet } from '../../../core/components/Fields';
import Button from '../../../core/components/Button';
import Heading from '../../../core/components/Heading';

const MSG = defineMessages({
  heading: {
    id: 'CreateColony.ColonyDetails.heading',
    defaultMessage: 'Create your very own Colony!',
  },
  buttonCreateColony: {
    id: 'CreateColony.ColonyDetails.button.createColony',
    defaultMessage: 'Create Colony',
  },
  helpText: {
    id: 'CreateColony.ColonyDetails.helpText',
    defaultMessage: 'So, this is some placeholder text',
  },
});

const displayName = 'dashboard.CreateColony.ColonyDetails';

type CustomProps = {
  nextStep: () => void,
};

type Props = FormProps<CustomProps>;

const ColonyDetails = ({ nextStep, handleSubmit, submitting }: Props) => (
  <section className={`${styles.main} ${layout.flexContent}`}>
    <header className={styles.header}>
      <Heading
        appearance={{ size: 'thin' }}
        text={MSG.heading}
      />
    </header>
    <form
      className="form"
      data-wd-hook="colony-creation-form"
      onSubmit={handleSubmit(nextStep)}
    >
      <FieldSet appearance={{ align: 'right' }}>
        <Button
          theme="primary"
          size="large"
          loading={submitting}
          value={MSG.buttonCreateColony}
        />
      </FieldSet>
    </form>
  </section>
);

ColonyDetails.displayName = displayName;

export default ColonyDetails;

export const helpText = <FormattedMessage {...MSG.helpText} />;
