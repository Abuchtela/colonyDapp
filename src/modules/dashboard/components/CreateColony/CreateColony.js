/* @flow */

import { compose } from 'recompose';

import withWizard from '../../../core/components/Wizard';
import CreateColony from './CreateColony.jsx';
import ColonyDetails, { helpText } from './ColonyDetails.jsx';

const steps = [
  { Step: ColonyDetails, sidebarChild: helpText },
];

const CreateColonyContainer = compose(withWizard({
  steps,
  form: 'create_colony',
}))(CreateColony);

export default CreateColonyContainer;
