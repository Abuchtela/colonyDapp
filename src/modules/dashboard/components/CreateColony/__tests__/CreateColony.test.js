/* eslint-env jest */

import React from 'react';
import toJson from 'enzyme-to-json';

import { shallowWithIntl } from '~utils/test';

import CreateColony from '../CreateColony.jsx';

describe('CreateColony component', () => {
  test('Renders initial component (snapshot)', () => {
    const wrapper = shallowWithIntl((
      <CreateColony />
    ));
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
