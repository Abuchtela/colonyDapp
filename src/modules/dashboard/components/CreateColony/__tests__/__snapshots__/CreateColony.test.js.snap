// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateColony component Renders initial component (snapshot) 1`] = `
<main
  className="undefined undefined undefined"
>
  <aside>
    <section>
      <figure
        role="presentation"
      >
        <svg
          viewBox="0 0 160 80"
        >
          <image
            height="80"
            href="test-file-stub"
            width="160"
            x="0"
            y="0"
          />
        </svg>
      </figure>
    </section>
  </aside>
  <article
    className="undefined undefined undefined undefined"
  />
</main>
`;
